test(integration): Flaky test TestScenario_SexModeDeactivation_ActivatesDayPhaseScene

The integration test `TestScenario_SexModeDeactivation_ActivatesDayPhaseScene` is flaky and fails intermittently in CI.

**Symptom:** Test times out waiting for scene activation after sex mode deactivation.

**Root cause:** The test uses `time.Sleep(50ms)` on line 273 to wait for async handlers before clearing service calls. This is insufficient in some environments.

**Example failure:**
```
--- FAIL: TestScenario_SexModeDeactivation_ActivatesDayPhaseScene (2.53s)
    scenario_sexmode_test.go:287: 
        Error:      	Condition never satisfied
        Test:       	TestScenario_SexModeDeactivation_ActivatesDayPhaseScene
        Messages:   	scene.primary_suite_sunset should be activated
```

**Suggested fix:** Replace `time.Sleep(50ms)` with polling helper (e.g., `waitForStringState` or similar) to ensure handlers complete before clearing service calls.

**Location:** `homeautomation-go/test/integration/scenario_sexmode_test.go:273`

**Impact:** Flaky CI failures unrelated to code changes (seen in PR #670).

--- a/homeautomation-go/test/integration/scenario_sexmode_test.go
+++ b/homeautomation-go/test/integration/scenario_sexmode_test.go
@@ -269,12 +269,12 @@ func TestScenario_SexModeDeactivation_ActivatesDayPhaseScene(t *testing.T) {
 
 	require.NoError(t, stateManager.SetString("dayPhase", "sunset"))
 	require.NoError(t, stateManager.SetBool("isMasterAsleep", false))
-	// Allow async handlers to process before clearing
-	time.Sleep(50 * time.Millisecond)
 
 	// Activate sex mode
 	server.SetState("input_boolean.sex", "on", nil)
 	waitForStringState(t, stateManager, "musicPlaybackType", "sex", "sex mode should activate")
+	// Wait for activation handler to fully complete (night scene is called before isActive is set)
+	waitForServiceCallWithEntity(t, server, "scene", "turn_on", "scene.primary_suite_night", "activation should complete with night scene")
 
 	server.ClearServiceCalls()
 
@@ -363,14 +363,12 @@ func TestScenario_SexModeDeactivation_DifferentDayPhases(t *testing.T) {
 
 			require.NoError(t, stateManager.SetString("dayPhase", phase))
 			require.NoError(t, stateManager.SetBool("isMasterAsleep", false))
-			// Allow async handlers to process before clearing
-			time.Sleep(50 * time.Millisecond)
 
 			// Activate sex mode
 			server.SetState("input_boolean.sex", "on", nil)
 			waitForStringState(t, stateManager, "musicPlaybackType", "sex", "sex mode should activate")
-			// Allow full handler completion (isActive is set after musicPlaybackType in handleSexModeOn)
-			time.Sleep(100 * time.Millisecond)
+			// Wait for activation handler to fully complete (night scene is called before isActive is set)
+			waitForServiceCallWithEntity(t, server, "scene", "turn_on", "scene.primary_suite_night", "activation should complete with night scene")
 
 			server.ClearServiceCalls()
 
@@ -411,8 +409,14 @@ func TestScenario_SexModeDuplicateActivation_Ignored(t *testing.T) {
 	// Activate sex mode
 	server.SetState("input_boolean.sex", "on", nil)
 	waitForStringState(t, stateManager, "musicPlaybackType", "sex", "sex mode should activate")
-	// Wait for full handler completion (isActive is set after musicPlaybackType in handleSexModeOn)
-	waitForCondition(t, func() bool { return sexModeManager.GetShadowState().Outputs.IsActive }, "sex mode should become active")
+	// Wait for night scene call (happens immediately before isActive is set in handleSexModeOn)
+	waitForServiceCallWithEntity(t, server, "scene", "turn_on", "scene.primary_suite_night", "Primary Suite night scene should activate during sex mode")
+	// Wait for all Eight Sleep climate calls to complete (these happen asynchronously)
+	assert.Eventually(t, func() bool {
+		calls := server.GetServiceCalls()
+		eightSleepCalls := FilterServiceCalls(calls, "climate", "set_temperature")
+		return len(eightSleepCalls) >= 2
+	}, stateWaitTimeout, statePollInterval, "Both Eight Sleep beds should be adjusted")
 
 	// Count initial service calls
 	initialCalls := len(server.GetServiceCalls())
@@ -603,8 +607,8 @@ func TestScenario_SexModeShadowState_TracksCorrectly(t *testing.T) {
 
 	server.SetState("input_boolean.sex", "on", nil)
 	waitForStringState(t, stateManager, "musicPlaybackType", "sex", "music should switch to sex")
-	// Wait for shadow state to update after state change
-	waitForCondition(t, func() bool { return sexModeManager.GetShadowState().Outputs.IsActive }, "shadow state should reflect activation")
+	// Brief delay for shadow state to update after state change
+	time.Sleep(50 * time.Millisecond)
 
 	t.Log("THEN: Shadow state should reflect activation")
 
@@ -620,8 +624,8 @@ func TestScenario_SexModeShadowState_TracksCorrectly(t *testing.T) {
 
 	server.SetState("input_boolean.sex", "off", nil)
 	waitForStringState(t, stateManager, "musicPlaybackType", "day", "music should be restored")
-	// Wait for shadow state to update after state change
-	waitForCondition(t, func() bool { return !sexModeManager.GetShadowState().Outputs.IsActive }, "shadow state should reflect deactivation")
+	// Brief delay for shadow state to update after state change
+	time.Sleep(50 * time.Millisecond)
 
 	t.Log("THEN: Shadow state should reflect deactivation")
 

